dipy/reconst/cti.py

```diff
@@ -0,0 +1,658 @@
+        gtab1: dipy.core.gradients.GradientTable
+        A GradientTable class instance for first DDE diffusion epoch
+        gtab2: dipy.core.gradients.GradientTable
+        A GradientTable class instance for second DDE diffusion epoch
+        S0 : float or ndarray (optional)
+            The non diffusion-weighted signal in every voxel, or across all
+            voxels. Default: 1
+
+        Returns
+        -------
+        S : numpy.ndarray
+            Signals.
+        """
+
+        return cti_prediction(cti_params, self.gtab1, self.gtab2, S0)
+
+
+class CorrelationTensorFit(DiffusionKurtosisFit):
+
+    """ Class for fitting the Correlation Tensor Model """
+
+    def __init__(self, model, model_params):
+        """ Initialize a CorrelationTensorFit class instance.
+
+        Since CTI is an extension of DKI, class instance is defined as subclass
+        of the DiffusionKurtosis from dki.py
+
+        Parameters
+        ----------
+        model : CorrelationTensorModel Class instance
+            Class instance containing the Correlation Tensor Model for the fit
+        model_params : ndarray (x, y, z, 48) or (n, 48)
+            All parameters estimated from the diffusion kurtosis model.
+            Parameters are ordered as follows:
+                1) Three diffusion tensor's eigenvalues
+                2) Three lines of the eigenvector matrix each containing the
+                   first, second and third coordinates of the eigenvector
+                3) Fifteen elements of the kurtosis tensor
+                4) Twenty One elements of the covariance tensor
+
+        """
+        DiffusionKurtosisFit.__init__(self, model, model_params)
+
+    @property
+    def kt(self):  # self.model_params.shape = (48, )
+        """
+        Return the 15 independent elements of the kurtosis tensor as an array
+        """
+        return self.model_params[12:27, ...]  
+
+    @property
+    def dft(self):  # created
+        """
+        Returns the 6 independent elements of the diffusion tensor as an array
+        """
+        # Extract the eigenvalues and the eigenvectors
+        evals = self.model_params[:3]
+        evecs = self.model_params[3:12].reshape((3, 3))
+
+        # Construct the diffusion tensor
+        diffusion_tensor = np.dot(np.dot(evecs, np.diag(evals)), evecs.T)
+
+        # Extract the independent elements of the diffusion tensor
+        dt_elements = np.array([diffusion_tensor[0, 0],
+                                diffusion_tensor[1, 1],
+                                diffusion_tensor[2, 2],
+                                diffusion_tensor[0, 1],
+                                diffusion_tensor[0, 2],
+                                diffusion_tensor[1, 2]])
+
+        return dt_elements
+    
+    @property
+    def cvt(self):  # created
+        """
+        Returns the 21 independent elements of the covariance tensor as an array
+        """
+        return self.model_params[27:48, ...]
```

Comment:
model parameters should correspond to the last dimension of model_params and not the first. 

return self.model_params[27:48, ...] ==> return self.model_params[..., 27:48]